### ISSUE TITLE
Delete `IndexedDbInterface`; just use the class instead

### ISSUE THREAD

**Issue by jessepinho** [penumbra-zone/web#1026]
Currently, when making updates to the `IndexedDb` class, we also have to make updates to `IndexedDbInterface`. This results in extra effort that doesn't seem to have any benefit: every place where `IndexedDbInterface` is used (except for [this test suite](https://github.com/penumbra-zone/web/blob/main/packages/query/src/price-indexer.test.ts#L9)) could just as easily use the class as its type instead.

This ticket is to delete the interface and just use the class.

**Comment by turbocrime** [comment]
there are many split class/interface definitions like this

**Comment by grod220** [comment]
I recall there was a potential issue with circular dependencies 🤔  and that the interface served as an indepedent dep. But perhaps this is no longer the case! Worth investigating.

**Comment by jessepinho** [comment]
@turbocrime: 

> there are many split class/interface definitions like this

Yeah, I've noticed that. I imagine that in some cases, it's to allow consumers to provide their own implementations but keep the same API contract. But in the case of IndexedDB, the interface is so tied to our own implementation that I can't imagine someone finding it useful.

@grod220:

> I recall there was a potential issue with circular dependencies

Interesting... I'll look into that if/when we pull this ticket into a sprint!

**Comment by TalDerei** [comment]
there was a circular dep here, closing as unplanned since it doesn't really yield any benefits.
